### PR TITLE
Create separate client libraries page on the documentation

### PR DESCRIPTION
**Pull request by filipecosta90**
The following PR addresses #544. 
- It's a simple move from the index.md to clients.md with no changes to the clients table.
- It also removes the Tests section from index.md given that testing is already addressed on the development.md page and it's really out of context on index.md

### CODE REVIEW

File: docs/clients.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# RedisTimeSeries Client Libraries
+
+RedisTimeSeries has several client libraries, written by the module authors and community members - abstracting the API in different programming languages. 
+
+While it is possible and simple to use the raw Redis commands API, in most cases it's easier to just use a client library abstracting it.
```
Comment (K-Jo) [inline docs/clients.md:5]: @filipecosta90 "it's more convenient"

**Review by K-Jo — CHANGES_REQUESTED** [review]
Can we add an extra column with completeness of API? @gkorland

File: docs/clients.md (hunk 1 above)
Comment (filipecosta90) [reply docs/clients.md]: Fixed in the latest commit. Please revise

**Comment by danni-m** [comment]
@filipecosta90 LGTM, but I think we need @K-Jo to merge this.

**Review by danni-m — DISMISSED** [review]

**Review by danni-m — APPROVED** [review]

**Pull request merged**